go/libraries/doltcore/dbfactory/grpc.go

```diff
@@ -36,7 +36,7 @@ var GRPCDialProviderParam = "__DOLT__grpc_dial_provider"
 
 // GRPCDialProvider is an interface for getting a *grpc.ClientConn.
 type GRPCDialProvider interface {
-	GetGRPCDialParams(grpcendpoint.Config) (string, []grpc.DialOption, error)
+	GetGRPCDialParams(grpcendpoint.Config) (string, []grpc.DialOption, grpcendpoint.HTTPFetcher, error)
```

Comment:
My main comment here is that 3 is generally the limit for return parameters, any more than that and you're straying into really hard to understand territory.

What about returning a struct / new interface that has needed methods on it?